sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -345,7 +161,106 @@ func GetRecoveryKind(ctxForLogging context.Context, err error) recoveryKind {
 		}
 	}
 
+	var me mgmtError
+
+	if errors.As(err, &me) {
+		code := me.RPCCode()
+
+		// this can happen when we're recovering the link - the client gets closed and the old link is still being
+		// used by this instance of the client. It needs to recover and attempt it again.
+		if code == 401 ||
+			// we lost the session lock, attempt link recovery
+			code == 410 {
+			return RecoveryKindLink
+		}
+
+		// simple timeouts
+		if me.Resp.Code == 408 || me.Resp.Code == 503 {
+			return RecoveryKindNone
+		}
+	}
+
 	// this is some error type we've never seen.
-	tab.For(ctxForLogging).Fatal(fmt.Sprintf("No recovery possible with error: %#v", err))
 	return RecoveryKindFatal
 }
+
+type (
+	// ErrMissingField indicates that an expected property was missing from an AMQP message. This should only be
+	// encountered when there is an error with this library, or the server has altered its behavior unexpectedly.
+	ErrMissingField string
+
+	// ErrMalformedMessage indicates that a message was expected in the form of []byte was not a []byte. This is likely
+	// a bug and should be reported.
+	ErrMalformedMessage string
```

Comment:
I know this is internal, but do we really need all these error types?  i.e. do we need to detect and handle them separately?

Reply:
I don't think there are any great spots where these would be used. Filed https://github.com/Azure/azure-sdk-for-go/issues/16896 to remove them.